homeassistant/components/nx584/alarm_control_panel.py

```diff
@@ -40,6 +43,7 @@
         vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_PARTITION, default=DEFAULT_PARTITION): cv.positive_int,
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Hi, to be honest I have almost no idea how to create a config flow, and that is after I read the info on the link you provided. I constructed the code based on another custom component so I am totally lost here.

Reply:
There's a scaffold script to get a template as a base to work with.

https://developers.home-assistant.io/docs/creating_component_index